x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/repository/CcrRepository.java

```diff
@@ -201,15 +204,40 @@ public void getSnapshotInfo(
         try {
             csDeduplicator.execute(
                 new ThreadedActionListener<>(threadPool.executor(ThreadPool.Names.SNAPSHOT_META), listener.map(response -> {
+                    Snapshot snapshot = new Snapshot(this.metadata.name(), SNAPSHOT_ID);
+
+                    // RestoreService will validate the snapshot is restorable based on the max index version.
+                    // However, we do not increment index version on patch releases.
+                    // To prevent attempting to restore an index of a future patch version, we reject the restore
+                    // already when building the snapshot info from newer nodes matching the current index version.
+                    IndexVersion maxIndexVersion = response.getNodes().getMaxDataNodeCompatibleIndexVersion();
+                    if (IndexVersion.current().id() == maxIndexVersion.id()) {
+                        for (var node : response.nodes()) {
+                            if (node.canContainData() && node.getMaxIndexVersion().id() == maxIndexVersion.id()) {
+                                BuildVersion remoteVersion = BuildVersion.fromVersionId(node.getVersion().id);
```

Comment:
This is using the discoverynode release version, which we want to remove at some point... If this needs to wait until we've worked out how to do CCR properly on serverless, can this have a comment indicating so?

Reply:
Is it really a serverless issue? In the case of serverless we're probably fine with index version only. I'd expect we would bump index version accordingly. IMHO, this is just to maintain the current version constraints for stateful.

Reply:
I've added a comment, in any case this needs to be revisited when looking into release version of discovery node